example_scripts/login_system-cache.pwn

```diff
@@ -46,224 +39,286 @@ new Player[MAX_PLAYERS][E_PLAYERS];
 new g_MysqlRaceCheck[MAX_PLAYERS];
 
 //dialog data
-enum 
+enum
 {
-	DIALOG_INVALID,
 	DIALOG_UNUSED,
-	
+
 	DIALOG_LOGIN,
-	DIALOG_REGISTER,
+	DIALOG_REGISTER
 };
 
-
-/*
- * SA-MP callbacks
- */
+main() {}
 
 public OnGameModeInit()
 {
-	mysql_log(LOG_ERROR | LOG_WARNING, LOG_TYPE_HTML); //logs errors and warnings into a nice .html log file
-	g_SQL = mysql_connect(SQL_HOST, SQL_USER, SQL_DB, SQL_PASS);
+	new MySQLOpt: option_id = mysql_init_options();
+
+	mysql_set_option(option_id, AUTO_RECONNECT, true); // It automatically reconnects when loosing connection to mysql server
+	mysql_set_option(option_id, MULTI_STATEMENTS, true); // Allows to execute multiple queries at once
```

Comment:
Yes, a single `mysql_tquery` call with multiple statements will produce multiple result sets, and as of R40 you can access a result set (or rather set it as active) with `cache_set_result(result_index);`.

SQL injections can also happen with multiple statements disabled (see my example above with the `%` input; `%` is a MySQL wildcard character). Only the possible harm is increased by a lot when enabling them.

Multiple statements is a MySQL feature, not a MySQL plugin feature. I just added the ability to enable/disable it.